### PR TITLE
Feature/77 deliverable definition api

### PR DESCRIPTION
**Pull request by shamoons**
Implements Create, Update and List API endpoints

### Summary

Uses proper express routing to CRU data

### Test Plan

Should b able to 

### Checklist

- [x ] Unit tests and linting pass
- [ ] Logging added, if applicable
- [ ] Code and logs checked for PHI/PII

### Security
N/A

### Additional JIRA Tickets (optional)

77, 78, 81

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -9,6 +10,59 @@ module.exports = (app, passport) => {
 
   })
 
+  app.post('/v1/dd', passport.authenticate('jwt', { session: false }), (req, res) => {
+    return global.db.DeliverableDefinition.create(req.body)
+      .then((resp) => {
+        res.json({
+          message: 'ok',
```
Comment (gopalakrishnanu) [inline routes/index.js]: Return the whole created object back, so the client does not have to make another call to get the object. Also the return code should be 201.

File: routes/index.js [hunk 2]
```diff
@@ -9,6 +10,59 @@ module.exports = (app, passport) => {
...
+          DeliverableDefinition_id: resp.id
+        })
+      })
+      .catch((err) => {
```
Comment (gopalakrishnanu) [inline routes/index.js]: Any validation errors should return 400 instead of 500

File: routes/index.js [hunk 3]
```diff
@@ -9,6 +10,59 @@ module.exports = (app, passport) => {
...
+
+  app.put('/v1/dd/:ddId', passport.authenticate('jwt', { session: false }), (req, res) => {
+    return global.db.DeliverableDefinition.update(req.body, {
+      where: {
+        id: req.params.ddId
+      }
+    })
+      .then((resp) => {
+        res.json({
+          message: 'ok'
```
Comment (gopalakrishnanu) [inline routes/index.js]: Return the updated object, instead of OK message,

**Comment by shamoons** [comment]
@gopalakrishnanu resolved

**Pull request merged**